flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/storage/FlintQueryCompiler.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.apache.spark.sql.flint.storage
+
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.connector.expressions.{Expression, FieldReference, LiteralValue}
+import org.apache.spark.sql.connector.expressions.filter.{And, Predicate}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+
+/**
+ * Todo. find the right package.
+ */
+case class FlintQueryCompiler(schema: StructType) {
+
+  /**
+   * Using AND to concat predicates. Todo. If spark spark.sql.ansi.enabled = true, more expression
+   * defined in V2ExpressionBuilder could be pushed down.
+   */
+  def compile(predicates: Array[Predicate]): String = {
+    if (predicates.isEmpty) {
+      return ""
+    }
+    compile(predicates.reduce(new And(_, _)))
```

Comment:
just to confirm, so Spark guarantees that all given predicates can be concat by AND? For IT `val df3 = df.filter($"aInt" === 1 || $"aString" === "b").sort(asc("aInt"))`, the `predicates` argument value will be a single predicate `aInt = 1 OR aString = b`?